File: 4. Java/EX58.java
package Apostila;

import java.util.Scanner;

public class EX58 
{
    public static void main(String[] args) 
    {
        // Declara variáveis para o número de linhas e colunas
        int m, n;

        // Usa try-with-resources para garantir que o Scanner será fechado
        try (Scanner ent = new Scanner(System.in)) 
        {
            // Solicita ao usuário o número de linhas da matriz
            System.out.print("# Quantas linhas vai ter cada matriz: ");
            m = protecao(ent);
            
            // Solicita ao usuário o número de colunas da matriz
            System.out.print("# Quantas colunas vai ter cada matriz: ");
            n = protecao(ent);
            
            // Declara matrizes A e B com tamanho m x n
            int [][] matA = new int[m][n];
            int [][] matB = new int[m][n];
            
            // Solicita os valores da matriz A
            System.out.println("\n# Digite os valores da matriz A: ");
            matA = popularMatriz(matA, ent);

            // Solicita os valores da matriz B
            System.out.println("\n# Digite os valores da matriz B: ");
            matB = popularMatriz(matB, ent);
            
            // Chama a função para somar as matrizes
            somaMatriz(matA, matB, m, n);
        }
    }
    
    // Função para proteger a entrada de dados
    private static int protecao(Scanner ent) 
    {
        // Variável para armazenar o valor lido
        int a = 0;
       
        // Loop para garantir que a entrada seja um número inteiro
        while (true) 
        {
            try
            {
                // Lê o valor inteiro
                a = ent.nextInt();
                break;
            } 

            catch (Exception e) 
            {
                // Solicita nova entrada em caso de erro
                System.out.print("# Digite numero inteiro: ");
                ent.nextLine(); // Limpa o buffer
            }
        }

        return a; // Retorna o valor lido
    }

    // Função para popular a matriz com valores do usuário
    private static int[][] popularMatriz(int mat[][], Scanner ent)
    {
        // Laço para percorrer as linhas da matriz
        for (int i = 0; i < mat.length; i++) 
        {
            // Laço para percorrer as colunas da matriz
            for (int j = 0; j < mat[i].length; j++)
            {
                // Solicita e lê o elemento na posição [i, j]
                System.out.printf("# Elemento [%d, %d]: ", i, j);
                
                // Verifica se a entrada é um inteiro
                while (!ent.hasNextInt())
                {
                    System.out.println("Entrada invalida. Tente novamente.");
                    ent.next(); // Descarte a entrada inválida
                }

                mat[i][j] = ent.nextInt(); // Armazena o valor lido
            }
        }

        return mat; // Retorna a matriz populada
    }
    
    // Função para somar duas matrizes
    private static void somaMatriz(int [][] mat1, int [][] mat2, int m, int n)
    {
        // Declara a matriz para armazenar a soma
        int [][] matC = new int[m][n];
        
        // Exibe o cabeçalho da matriz soma
        System.out.println("\n# MATRIZ SOMA:");
        
        // Laço para percorrer as linhas da matriz de soma
        for (int i = 0; i < matC.length; i++)
        {                    
            // Laço para percorrer as colunas da matriz de soma
            for (int j = 0; j < matC[i].length; j++)
            {
                // Calcula a soma dos elementos correspondentes
                matC[i][j] = mat1[i][j] + mat2[i][j];
                
                // Exibe o elemento somado
                System.out.print(matC[i][j] + " ");
            }

            // Nova linha após cada linha da matriz soma
            System.out.println(); // Formata a saída
        }
        
        // Linha em branco após a exibição da matriz soma
        System.out.println("");
    }
}

File: 6. Python/EX58.py
def popular_matriz(matriz, tipo, txt):

    # Solicita os valores para preencher a matriz
    print(f"\n# Digite os valores da {txt}: ")

    for i in range(len(matriz)):
        for j in range(len(matriz[0])):
            while True:

                try:
                    matriz[i][j] = tipo(input(f"# Elemento [{i},{j}]: "))  # Captura o elemento
                    break

                except Exception:
                    print("\n# Entrada invalida !")  # Mensagem de erro

def matriz_soma(matriz1, matriz2):

    # Cria a matriz resultante da soma
    matC = [[0 for _ in range(len(matriz1[0]))] for _ in range(len(matriz1))]

    print("\n# MATRIZ SOMA: ", end="")
    for i in range(len(matC)):
        print("")

        for j in range(len(matC[0])):
            matC[i][j] = matriz1[i][j] + matriz2[i][j]  # Soma as matrizes
            print(f"{matC[i][j]} ", end="")             # Imprime resultado

def protecao_entrada_dados(txt, tipo):

    # Valida a entrada de dados
    while True:

        try:
            val = tipo(input(txt))  # Captura valor
            if val > 0:
                return val  # Retorna valor válido

        except Exception:
            print("\n* Entrada invalida !")  # Mensagem de erro

def main():

    # Solicita dimensões das matrizes
    m = protecao_entrada_dados("# Quantas linhas vai ter cada matriz: ", int)
    n = protecao_entrada_dados("# Quantas colunas vai ter cada matriz: ", int)

    # Inicializa matrizes
    matA = [[0 for _ in range(n)] for _ in range(m)]
    matB = [[0 for _ in range(n)] for _ in range(m)]

    # Preenche as matrizes
    popular_matriz(matA, int, "matriz A")
    popular_matriz(matB, int, "matriz B")

    # Soma as matrizes
    matriz_soma(matA, matB)

if __name__ == '__main__':
    main()  # Executa o programa